config/config.go

```diff
@@ -7,77 +7,96 @@ import (
 
 	"github.com/mitchellh/go-homedir"
 	"gopkg.in/yaml.v2"
+	"knative.dev/func/builders"
+	"knative.dev/func/k8s"
+	"knative.dev/func/openshift"
 )
 
 const (
 	// Filename into which Config is serialized
 	Filename = "config.yaml"
 
-	// DefaultConfigPath is used in the unlikely event that
-	// the user has no home directory (no ~), there is no
-	// XDG_CONFIG_HOME set
-	DefaultConfigPath = ".config/func"
+	// Repositories is the default directory for repositoires.
+	Repositories = "repositories"
 
 	// DefaultLanguage is intentionaly undefined.
 	DefaultLanguage = ""
+
+	// DefaultBuilder defined by builders
+	DefaultBuilder = builders.Default
 )
 
-type Config struct {
-	// Language Runtime
-	Language string `yaml:"language"`
+func DefaultRegistry() string {
+	switch {
+	case openshift.IsOpenShift():
+		return openshift.GetDefaultRegistry()
+	default:
+		return ""
+	}
+}
+
+// DefaultNamespace for remote operations is the currently active
+// context namespace (if available) or the fallbacl "default".
```

Comment:
Nit: (surprised reviewdog did not catch this)
```suggestion
// context namespace (if available) or the fallback "default".
```